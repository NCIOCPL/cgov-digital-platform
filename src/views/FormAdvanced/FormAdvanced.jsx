import React from 'react';
import PropTypes from 'prop-types';

import {
  CancerTypeCondition,
  Age,
  KeywordsPhrases,
  Location,
  TrialType,
  DrugTreatment,
  TrialPhase,
  TrialId,
  TrialInvestigators,
  LeadOrganization,
} from '../../components/search-modules';

const FormAdvanced = () => {
  const handleSubmit = e => {
    e.preventDefault();
    console.log('form submitted');
  };

  return (
    <form onSubmit={handleSubmit} className="search-page__form advanced">
      <CancerTypeCondition />
      <Age />
      <KeywordsPhrases />
      <Location />
      <TrialType />
      <DrugTreatment />
      <TrialPhase />
      <TrialId />
      <TrialInvestigators />
      <LeadOrganization />
    </form>
  );
};

FormAdvanced.propTypes = {};

FormAdvanced.defaultProps = {};

export default FormAdvanced;
