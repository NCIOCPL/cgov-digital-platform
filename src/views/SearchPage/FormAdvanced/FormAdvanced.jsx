import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import {
  CancerTypeCondition,
  Age,
  KeywordsPhrases,
  Location,
  TrialType,
  DrugTreatment,
  TrialPhase,
  TrialId,
  TrialInvestigators,
  LeadOrganization,
} from '../../../components/search-modules';

const FormAdvanced = ({ handleUpdate }) => {
  //Module groups in arrays will be placed side-by-side in the form
  const modules = [
    CancerTypeCondition,
    [Age, KeywordsPhrases],
    Location,
    TrialType,
    DrugTreatment,
    TrialPhase,
    TrialId,
    TrialInvestigators,
    LeadOrganization,
  ];

  return (
    <>
      {modules.map((Module, idx) => {
        if (Array.isArray(Module)) {
          return <div key={`formAdvanced-${idx}`} className="side-by-side">
            {Module.map((Mod, i) => <Mod key={`formAdvanced-${idx}-${i}`} handleUpdate={handleUpdate} />)}
          </div>;
        } else {
          return <Module key={`formAdvanced-${idx}`} handleUpdate={handleUpdate} />;
        }
      })}
    </>
  );
};

FormAdvanced.propTypes = {
  handleUpdate: PropTypes.func,
};

export default FormAdvanced;