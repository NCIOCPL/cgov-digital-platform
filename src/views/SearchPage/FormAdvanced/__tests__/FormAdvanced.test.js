import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import FormAdvanced from '../FormAdvanced';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <FormAdvanced {...props} />
  );
  return component;
};

describe('FormAdvanced', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
