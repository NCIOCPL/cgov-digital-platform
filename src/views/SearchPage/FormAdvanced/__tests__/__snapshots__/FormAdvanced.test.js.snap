// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAdvanced Render renders without error 1`] = `
<form
  class="search-page__form advanced"
>
  <fieldset
    class="cts-fieldset "
    id="fieldset--type"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Cancer Type/Condition
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#cancertype"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <p>
        Select a cancer type or condition. Select additional options, if applicable.
      </p>
      <div
        class=""
      >
        <label
          class="cts-label"
          for="q"
          id="q-label"
        >
          Primary Cancer Type/Condition
        </label>
        <input
          aria-labelledby="q-label"
          aria-required="false"
          class="cts-input  "
          id="q"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      >
        <label
          class="cts-label"
          for="st"
          id="st-label"
        >
          Subtype
        </label>
        <input
          aria-labelledby="st-label"
          aria-required="false"
          class="cts-input  "
          id="st"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      >
        <label
          class="cts-label"
          for="stage"
          id="stage-label"
        >
          Stage
        </label>
        <input
          aria-labelledby="stage-label"
          aria-required="false"
          class="cts-input  "
          id="stage"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      >
        <label
          class="cts-label"
          for="fin"
          id="fin-label"
        >
          Side Effects/Biomarkers/Participant Attributes
        </label>
        <input
          aria-labelledby="fin-label"
          aria-required="false"
          class="cts-input  "
          id="fin"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </fieldset>
  <div
    class="side-by-side"
  >
    <fieldset
      class="cts-fieldset "
      id="fieldset--age"
    >
      <legend
        class="cts-fieldset__legend"
      >
        <span>
          Age
        </span>
      </legend>
      <a
        aria-label="Help"
        class="text-icon-help"
        href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
        rel="noopener noreferrer"
        target="_blank"
      >
        ?
      </a>
      <div
        class="cts-fieldset__body"
      >
        <div
          class=""
        >
          <label
            class="cts-label"
            for="a"
            id="a-label"
          >
            Enter the age of the participant.
          </label>
          <input
            aria-labelledby="a-label"
            aria-required="false"
            class="cts-input  "
            id="a"
            maxlength="3"
            spellcheck="false"
            type="text"
            value=""
          />
        </div>
      </div>
    </fieldset>
    <fieldset
      class="cts-fieldset "
      id="fieldset--keyword"
    >
      <legend
        class="cts-fieldset__legend"
      >
        <span>
          Keywords/Phrases
        </span>
      </legend>
      <a
        aria-label="Help"
        class="text-icon-help"
        href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#keywords"
        rel="noopener noreferrer"
        target="_blank"
      >
        ?
      </a>
      <div
        class="cts-fieldset__body"
      >
        <div
          class=""
        >
          <label
            class="cts-label"
            for="q"
            id="q-label"
          >
            Search by word or phrase (use quotation marks with phrases).
          </label>
          <input
            aria-labelledby="q-label"
            aria-required="false"
            class="cts-input  "
            id="q"
            placeholder="Examples: PSA, 'Paget disease'"
            spellcheck="false"
            type="text"
            value=""
          />
        </div>
      </div>
    </fieldset>
  </div>
  <fieldset
    class="cts-fieldset search-location"
    id="fieldset--location"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Location
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#location"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <p>
        Search for trials near a specific zip code; or in a country, state and city; or at a particular institution. The default selection will search for trials in all available locations.
      </p>
      <div
        class="data-toggle-block"
      >
        <div
          class="cts-toggle "
        >
          <input
            class="cts-toggle__input"
            id="search-location-toggle"
            name="search-location-toggle"
            type="checkbox"
          />
          <label
            aria-label="Limit results to Veterans Affairs facilities"
            class="cts-toggle__label"
            for="search-location-toggle"
          >
            <span
              aria-hidden="true"
              class="neg"
            >
              No
            </span>
            <span
              aria-hidden="true"
              class="pos"
            >
              Yes
            </span>
          </label>
        </div>
        Limit results to Veterans Affairs facilities
      </div>
      <div
        class="group-locations"
      >
        <div
          class="cts-radio "
        >
          <input
            aria-disabled="false"
            class="cts-radio__input"
            id="search-location-all"
            name="radios"
            type="radio"
            value="search-location-all"
          />
          <label
            class="cts-radio__label"
            for="search-location-all"
          >
            Search All Locations
          </label>
        </div>
        <div
          class="cts-radio "
        >
          <input
            aria-disabled="false"
            class="cts-radio__input"
            id="search-location-zip"
            name="radios"
            type="radio"
            value="search-location-zip"
          />
          <label
            class="cts-radio__label"
            for="search-location-zip"
          >
            ZIP Code
          </label>
        </div>
        <div
          class="cts-radio "
        >
          <input
            aria-disabled="false"
            class="cts-radio__input"
            id="search-location-country"
            name="radios"
            type="radio"
            value="search-location-country"
          />
          <label
            class="cts-radio__label"
            for="search-location-country"
          >
            Country, State, City
          </label>
        </div>
        <div
          class="cts-radio "
        >
          <input
            aria-disabled="false"
            class="cts-radio__input"
            id="search-location-hospital"
            name="radios"
            type="radio"
            value="search-location-hospital"
          />
          <label
            class="cts-radio__label"
            for="search-location-hospital"
          >
            Hospitals/Institutions
          </label>
        </div>
        <div
          class="cts-radio "
        >
          <input
            aria-disabled="false"
            class="cts-radio__input"
            id="search-location-nih"
            name="radios"
            type="radio"
            value="search-location-nih"
          />
          <label
            class="cts-radio__label"
            for="search-location-nih"
          >
            At NIH (only show trials at the NIH clinical center in Bethesda, MD)
          </label>
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="cts-fieldset trial-type"
    id="fieldset--trialtype"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Trial Type
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialtype"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <p>
        Select the type of trial for your search. You may check more than one box or select "All."
      </p>
      <div
        class="data-toggle-block"
      >
        <div
          class="cts-toggle "
        >
          <input
            class="cts-toggle__input"
            id="hv"
            name="hv"
            type="checkbox"
          />
          <label
            aria-label="Limit results to Veterans Affairs facilities"
            class="cts-toggle__label"
            for="hv"
          >
            <span
              aria-hidden="true"
              class="neg"
            >
              No
            </span>
            <span
              aria-hidden="true"
              class="pos"
            >
              Yes
            </span>
          </label>
        </div>
        Limit results to trials accepting healthy volunteers
      </div>
      <div
        class="select-all"
      >
        <div
          class="cts-checkbox tt-all"
        >
          <input
            checked=""
            class="cts-checkbox__input"
            id="tt_all"
            name="tt"
            type="checkbox"
            value="tt_all"
          />
          <label
            class="cts-checkbox__label"
            for="tt_all"
          >
            All
          </label>
        </div>
      </div>
      <div
        class="group-trial-types"
      >
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_treatment"
            name="tt"
            type="checkbox"
            value="treatment"
          />
          <label
            class="cts-checkbox__label"
            for="tp_treatment"
          >
            Treatment
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_prevention"
            name="tt"
            type="checkbox"
            value="prevention"
          />
          <label
            class="cts-checkbox__label"
            for="tp_prevention"
          >
            Prevention
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_supportive_care"
            name="tt"
            type="checkbox"
            value="supportive_care"
          />
          <label
            class="cts-checkbox__label"
            for="tp_supportive_care"
          >
            Supportive Care
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_health_services_research"
            name="tt"
            type="checkbox"
            value="health_services_research"
          />
          <label
            class="cts-checkbox__label"
            for="tp_health_services_research"
          >
            Health Services Research
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_diagnostic"
            name="tt"
            type="checkbox"
            value="diagnostic"
          />
          <label
            class="cts-checkbox__label"
            for="tp_diagnostic"
          >
            Diagnostic
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_screening"
            name="tt"
            type="checkbox"
            value="screening"
          />
          <label
            class="cts-checkbox__label"
            for="tp_screening"
          >
            Screening
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_basic_science"
            name="tt"
            type="checkbox"
            value="basic_science"
          />
          <label
            class="cts-checkbox__label"
            for="tp_basic_science"
          >
            Basic Science
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_other"
            name="tt"
            type="checkbox"
            value="other"
          />
          <label
            class="cts-checkbox__label"
            for="tp_other"
          >
            Other
          </label>
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="cts-fieldset "
    id="fieldset--type"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Drug/Treatment
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#drugtreatment"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <p>
        Search for a specific drug or intervention. You can use the drug's generic or brand name.
      </p>
      <div
        class="cts-autocomplete "
        id="dt-autocomplete-wrapper"
      >
        <label
          class="cts-label"
          for="dt"
          id="dt-label"
        >
          Drug/DrugFamily
        </label>
        <div
          class="cts-chip-list"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            autocomplete="off"
            class="cts-input cts-autocomplete__input"
            id="dt"
            role="combobox"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=""
      >
        <label
          class="cts-label"
          for="ot"
          id="ot-label"
        >
          Other Treatments
        </label>
        <input
          aria-labelledby="ot-label"
          aria-required="false"
          class="cts-input  "
          id="ot"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </fieldset>
  <fieldset
    class="cts-fieldset trial-phase"
    id="fieldset--trialphase"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Trial Phase
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialphase"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <p>
        Search by one or more trial phases.
      </p>
      <div
        class="select-all"
      >
        <div
          class="cts-checkbox tp-all"
        >
          <input
            checked=""
            class="cts-checkbox__input"
            id="tp_all"
            name="tp"
            type="checkbox"
            value="tp_all"
          />
          <label
            class="cts-checkbox__label"
            for="tp_all"
          >
            All
          </label>
        </div>
      </div>
      <div
        class="group-phases"
      >
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_I"
            name="tp"
            type="checkbox"
            value="I"
          />
          <label
            class="cts-checkbox__label"
            for="tp_I"
          >
            Phase I
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_II"
            name="tp"
            type="checkbox"
            value="II"
          />
          <label
            class="cts-checkbox__label"
            for="tp_II"
          >
            Phase I
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_III"
            name="tp"
            type="checkbox"
            value="III"
          />
          <label
            class="cts-checkbox__label"
            for="tp_III"
          >
            Phase III
          </label>
        </div>
        <div
          class="cts-checkbox "
        >
          <input
            class="cts-checkbox__input"
            id="tp_IV"
            name="tp"
            type="checkbox"
            value="IV"
          />
          <label
            class="cts-checkbox__label"
            for="tp_IV"
          >
            Phase IV
          </label>
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="cts-fieldset "
    id="fieldset--trialid"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Trial ID
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialid"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <div
        class=""
      >
        <label
          class="cts-label"
          for="tid"
          id="tid-label"
        >
          Separate multiple IDs with commas or semicolons.
        </label>
        <input
          aria-labelledby="tid-label"
          aria-required="false"
          class="cts-input  "
          id="tid"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </fieldset>
  <fieldset
    class="cts-fieldset "
    id="fieldset--trialinvestigator"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Trial Investigators
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialinvestigators"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <div
        class="cts-autocomplete "
        id="component-unique-id-1-autocomplete-wrapper"
      >
        <label
          class="cts-label"
          for="component-unique-id-1"
          id="component-unique-id-1-label"
        >
          Search by Trial Investigators
        </label>
        <div
          class=""
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            autocomplete="off"
            class="cts-input cts-autocomplete__input"
            id="component-unique-id-1"
            role="combobox"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="cts-fieldset "
    id="fieldset--lead_organization"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Lead Organization
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#leadorganization"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <div
        class="cts-autocomplete "
        id="component-unique-id-2-autocomplete-wrapper"
      >
        <label
          class="cts-label"
          for="component-unique-id-2"
          id="component-unique-id-2-label"
        >
          Search by Lead Organization
        </label>
        <div
          class=""
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            autocomplete="off"
            class="cts-input cts-autocomplete__input"
            id="component-unique-id-2"
            role="combobox"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </fieldset>
  <div
    class="submit-block"
  >
    <button
      class="btn-submit"
      type="submit"
    >
      Find Trials
    </button>
  </div>
</form>
`;
